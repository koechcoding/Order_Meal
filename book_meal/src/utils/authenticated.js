export default function authenticated(){
    try{
        return JSON.parse(localStorage.getItem('user'));
    } catch(err){
        
    }
}