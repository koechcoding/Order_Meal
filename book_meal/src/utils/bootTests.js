import Enzyme from "enzyme";
import Adapter from "enzyme-adapter-react-16";
Enzyme.configure({ adapter: new Adapter()});

global.window = {}
import localStorage from 'mock-local-Storage';
window.localStorage = global.localStorage

export default Enzyme;
