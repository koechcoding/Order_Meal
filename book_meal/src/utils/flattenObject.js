export default function flattenObject(obj){
    let result = {}
    for (const key in obj){
        if((typeof obj[key])=== 'object' &&  obj[key]  !== null) {
            let flattened = flattenObject(obj[key])
            for(const key2 in flattened){
                result[key + "." + key2] = flattened[key2];
            }
        }else{
            result[key] = obj[key];
        }
    }
    return result;
}