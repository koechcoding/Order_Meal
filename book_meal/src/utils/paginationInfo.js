export default function paginationInfo(page){

    if(!page){
        return{}
    }

    let currentPage = 1
    if(page.next_page)
         currentPage = page.next_page -1;
    else if (page.prev_page)
        currentPage = page.prev_page + 1;

    let pageCount = Math.ceil(page.total / page.per_paage);
    if(Number.isNaN(pageCount) || pageCount === 0)
      pageCount = 1; 

    return {
        pageCount,
        currentPage,
        total: page.total,
        perPage: page.per_page,
        hasNext: page.has_next,
        hasPrev: page.has_prev,
        nextPage: page.next_page,
        prevPage: page.prev_page,
        currentCount: page.current_count,
    };
}