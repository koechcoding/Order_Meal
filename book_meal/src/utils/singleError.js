const singleError = (errorPayload)=>{
    if(errorPayload && errorPayload.data){
        const response = errorPayload.data;
        //alternative to Object.keys...
        for(let errorKey in response.errors){
            return response.error[errorKey]
        }
        if(response.message){
            return response.message;
        }
    }
    return 'Unknown error. Please try again!';
}
export default singleError;