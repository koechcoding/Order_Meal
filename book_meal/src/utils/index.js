import singleError from './singleError';
import paginationInfo from './paginationInfo';
import authenticated from './authenticated';
import flattenObject from './flattenObject';

export { 
    singleError, 
    paginationInfo, 
    authenticated,
    flattenObject
}