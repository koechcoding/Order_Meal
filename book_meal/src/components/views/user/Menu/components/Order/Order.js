import React from 'react';
import PropTypes from 'prop-types';
import axios from 'src/axios';
import Modal from 'src/components/common/Modal';
import { Alert, Button, Input } from 'reactstrap';
import { authenticated, singleError } from 'src/utils';


class OrderModal extends React.Component {

    state = {
        quantity: 1
    }

    reset = () => {
        setTimeout(() => {
            this.setState({
                ...this.state,
                success: false,
                error: null
            });
        }, 2000);
    }

    onOpened = () => {
        this.setState({
            ...this.state,
            error: null,
            success: false,
        });
    }

    onOrder = () => {
        const user = authenticated();
        const { item } = this.props;
        const order = {
            user_id: user.id,
            menu_item_id: item.id,
            quantity: Number(this.state.quantity)
        }
        this.props.setLoading(true);

        axios.post('orders', order).then(({ data }) => {
            this.props.setLoading(false);
            this.setState({
                ...this.state,
                success: true,
            });
            setTimeout(this.props.toggle, 1000);
        }).catch(({ response }) => {
            this.setState({
                ...this.state,
                error: response
            });
            this.props.setLoading(false);
        });
    }

    onChange = (e) => {
        this.setState({
            ...this.state,
            [e.target.name]: e.target.value
        });
    }

    render() {
        const { error, success } = this.state;
        const { item = {}, menu = {} } = this.props;
        const { meal = {} } = item;
        const body = (
            <div>
                {error &&
                    <Alert className="text-center text-small" color="danger">
                        { singleError(error) }
                    </Alert>
                }
                {success &&
                    <Alert className="text-center text-small" color="success">
                        Successfully made order.
                    </Alert>
                }
                <p className="text-center">Order for {meal.name} on {menu.name}?</p>
                <label>Quantity</label>
                <Input type="number" 
                    name="quantity"
                    value={this.state.quantity} 
                    onChange={this.onChange} />
            </div>
        );

        const footer = (
            <Button color="primary" className="m-auto" onClick={this.onOrder}>Make Order</Button>
        );

        const { isOpen, toggle } = this.props;
        return (
            <Modal 
                {...this.props}
                title="Make An Order" 
                body={body} 
                footer={footer} 
                isOpen={isOpen}
                toggle={toggle}
                onOpened={this.onOpened}
            />
        );
    }
}

OrderModal.propTypes = {
    setLoading: PropTypes.func.isRequired,
    toggle: PropTypes.func.isRequired,
    item: PropTypes.object,
    menu: PropTypes.object,
}

export default OrderModal;