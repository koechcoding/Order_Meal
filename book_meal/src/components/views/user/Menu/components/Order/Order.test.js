import React from 'react';
import { shallow }  from 'enzyme';
import 'src/utils/bootTests'; 
import Order from './Order';

const props = {toggle: () => {}, setLoading: () => {}, item: {}, menu: {}}
describe('<Order />', () => {
    it('renders correctly', () => {
        const wrapper = shallow(<Order {...props}/>);
        expect(wrapper).toMatchSnapshot();
    });
    it('resets on opened', () => {
        const wrapper = shallow(<Order {...props}/>);
        wrapper.setState({...wrapper.state(), error: 'error'});
        wrapper.instance().onOpened();
        expect(wrapper.state('error')).toBe(null);
    });
    it('records input changes', () => {
        const wrapper = shallow(<Order {...props}/>);
        wrapper.instance().onChange({target:{name: 'name', value: 'value'}});
        expect(wrapper.state('name')).toBe('value');
    });
});