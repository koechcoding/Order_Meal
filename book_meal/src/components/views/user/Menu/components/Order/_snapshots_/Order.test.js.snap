// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Order /> renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <OrderModal
    item={Object {}}
    menu={Object {}}
    setLoading={[Function]}
    toggle={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "body": <div>
        <p
                className="text-center"
        >
                Order for 
                 on 
                ?
        </p>
        <label>
                Quantity
        </label>
        <Input
                name="quantity"
                onChange={[Function]}
                type="number"
                value={1}
        />
</div>,
      "footer": <Button
        className="m-auto"
        color="primary"
        onClick={[Function]}
        tag="button"
>
        Make Order
</Button>,
      "isOpen": undefined,
      "item": Object {},
      "menu": Object {},
      "onOpened": [Function],
      "setLoading": [Function],
      "title": "Make An Order",
      "toggle": [Function],
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "body": <div>
          <p
                    className="text-center"
          >
                    Order for 
                     on 
                    ?
          </p>
          <label>
                    Quantity
          </label>
          <Input
                    name="quantity"
                    onChange={[Function]}
                    type="number"
                    value={1}
          />
</div>,
        "footer": <Button
          className="m-auto"
          color="primary"
          onClick={[Function]}
          tag="button"
>
          Make Order
</Button>,
        "isOpen": undefined,
        "item": Object {},
        "menu": Object {},
        "onOpened": [Function],
        "setLoading": [Function],
        "title": "Make An Order",
        "toggle": [Function],
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;